modules/RTC/TraceablePeerConnection.js

```diff
@@ -166,6 +210,195 @@ var dumpSDP = function(description) {
 };
 
 /**
+ * Called when new remote MediaStream is added to the PeerConnection.
+ * @param {MediaStream} stream the WebRTC MediaStream for remote participant
+ */
+TraceablePeerConnection.prototype._remoteStreamAdded = function (stream) {
+    const self = this;
+    if (!RTC.isUserStream(stream)) {
+        logger.info(
+            "Ignored remote 'stream added' event for non-user stream", stream);
+        return;
+    }
+    // Bind 'addtrack'/'removetrack' event handlers
+    if (RTCBrowserType.isChrome() || RTCBrowserType.isNWJS()
+        || RTCBrowserType.isElectron()) {
+        stream.onaddtrack = function (event) {
+            self._remoteTrackAdded(event.target, event.track);
+        };
+        stream.onremovetrack = function (event) {
+            self._remoteTrackRemoved(event.target, event.track);
+        };
+    }
+    // Call remoteTrackAdded for each track in the stream
+    stream.getAudioTracks().forEach(function (track) {
+        self._remoteTrackAdded(stream, track);
+    });
+    stream.getVideoTracks().forEach(function (track) {
+        self._remoteTrackAdded(stream, track);
+    });
+};
+
+
+/**
+ * Called on "track added" and "stream added" PeerConnection events(cause we
+ * handle streams on per track basis). Does find the owner and the SSRC for
+ * the track and passes that to ChatRoom for further processing.
+ * @param {MediaStream} stream the WebRTC MediaStream instance which is
+ * the parent of the track
+ * @param {MediaStreamTrack} track the WebRTC MediaStreamTrack added for remote
+ * participant
+ */
+TraceablePeerConnection.prototype._remoteTrackAdded = function (stream, track) {
+    const streamId = RTC.getStreamID(stream);
+    const mediaType = track.kind;
+
+    logger.info("Remote track added", streamId, mediaType);
+
+    // look up an associated JID for a stream id
+    if (!mediaType) {
+        GlobalOnErrorHandler.callErrorHandler(
+            new Error(
+                "MediaType undefined for remote track, stream id: "
+                    + streamId));
+        // Abort
```

Comment:
It doesn't hurt. I only pointed them out because IMHO they don't add much, since the reader can see the function will return early. No strong opinion, I leave it up to you.